timelock-server/src/test/java/com/palantir/atlasdb/timelock/lock/AsyncLockServiceEteTest.java

```diff
@@ -119,8 +117,8 @@ public void waitingRequestGetsTheLockAfterItIsUnlockedWithMultipleLocks() {
     public void requestsAreIdempotentDuringAcquisitionPhase() {
         LockToken currentHolder = lockSynchronously(REQUEST_1, LOCK_A);
 
-        AsyncResult<LockToken> tokenResult = lock(REQUEST_2, LOCK_A);
-        AsyncResult<LockToken> duplicateResult = lock(REQUEST_2, LOCK_A);
+        AsyncResult<LeasableLockToken> tokenResult = lock(REQUEST_2, LOCK_A);
```

Comment:
as noted before; I think you can do with some tests that validate some stuff to do with leasing; that a lock comes back with a lease, that the lease times make sense, that the lease duration makes sense, etc.

Reply:
Like, we currently have a random constant; if that constant were > 20 seconds there'd be a correctness issue here; we should have a test that would catch that correctness issue.